website/common/script/libs/taskDefaults.js

```diff
@@ -9,7 +9,7 @@ import moment from 'moment';
 
 const tasksTypes = ['habit', 'daily', 'todo', 'reward'];
 
-module.exports = function taskDefaults (task = {}) {
+module.exports = function taskDefaults (task = {}, user = { preferences: { dayStart: 0 } }) {
```

Comment:
The default user parameter works well but I think there should be a check instead that throws an error is the user object isn't passed since in that case the startDate would be incorrect

Reply:
What's the best way to raise an error in this context? Something like:

```javascript
throw new Error('No user passed to taskDefaults.')
```
(I come from Python, so still figuring out the JavaScript equivalents.)

Or perhaps it would be better to remove default arguments all together (since `taskDefaults` is never called without a task in the current codebase).

Reply:
Actually yeah I think removing the default parameter should be enough as it'll throw an error automatically if no user is passed